sim/midas/src/main/scala/midas/passes/fame/EmitAndWrapRAMModels.scala

```diff
@@ -226,18 +226,21 @@ class EmitAndWrapRAMModels extends Transform {
       val readWritePorts = annos.collect({ case anno: ModelReadWritePort  => new ReadWritePort(anno, mod.ports)})
       require(readWritePorts.isEmpty)
 
-      val clocks = mod.ports.filter(_.tpe == ClockType)
-      // TODO: turn this back to == 1
-      assert(clocks.length >= 1)
-      val hostClock = clocks.find(_.name == "clock").getOrElse(clocks.head) // TODO: naming convention for host clock
+      val (hostClock, updatedPortList) = mod.ports.find(p => p.tpe == ClockType && p.name == "clock") match {
+        case Some(port) => (port, mod.ports)
+        case None =>
```

Comment:
So this is just checking if the extracted model (not the generated RAM model) has a port named `clock`? And doesn't `hostClock` contain the same port in either case?